osu.Game/Screens/Edit/Compose/Components/BlueprintContainer.cs

```diff
@@ -117,19 +130,34 @@ protected override bool OnClick(ClickEvent e)
             return true;
         }
 
+        protected override void Update()
+        {
+            base.Update();
+
+            if (currentPlacement != null)
+            {
+                bool cursorInPlayfield = composer.RulesetContainer.Playfield.ReceivePositionalInputAt(inputManager.CurrentState.Mouse.Position);
```

Comment:
as discussed, too much chaining. make a local property on composer which does the rest of the chain (probably virtual so the consumer could customise it).****